braindecode/datasets/datasets.py

```diff
@@ -0,0 +1,446 @@
+        split_ids.update({group_name: list(group.index)})
+    return split_ids
+
+
+class BNCI2014001(MOABBDataset):
+    """
+    see moabb.datasets.bnci.BNCI2014001
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__("BNCI2014001", *args, **kwargs)
+
+
+class HGD(MOABBDataset):
+    """
+    see moabb.datasets.schirrmeister2017.Schirrmeister2017
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__("Schirrmeister2017", *args, **kwargs)
+
+
+class TUHAbnormal(ConcatDataset):
+    """
+    Temple University Hospital (TUH) Abnormal EEG Corpus.
+
+    Parameters
+    ----------
+    trial_start_offset_samples: int
+        start offset from original trial onsets in samples
+    trial_stop_offset_samples: int
+        stop offset from original trial onsets in samples
+    supercrop_size_samples: int
+        supercrop size
+    supercrop_stride_samples: int
+        stride between supercrops
+    drop_samples: bool
+        whether or not have a last overlapping supercrop/window, when
+        supercrops/windows do not equally devide the continuous signal
+    target: str
+
+    mapping: dict{target_value: int}
+        maps target values to integers
+    """
+
+    def __init__(self, path, trial_start_offset_samples,
+                 trial_stop_offset_samples, supercrop_size_samples,
+                 supercrop_stride_samples, subject_ids=None,
+                 drop_samples=False, target="pathological", mapping=None):
+        windower = FixedLengthWindower(
+            trial_start_offset_samples=trial_start_offset_samples,
+            trial_stop_offset_samples=trial_stop_offset_samples,
+            supercrop_size_samples=supercrop_size_samples,
+            supercrop_stride_samples=supercrop_stride_samples,
+            drop_samples=drop_samples, mapping=mapping)
+
+        all_file_paths = read_all_file_names(
+            path, extension='.edf', key=self._time_key)
+        if subject_ids is None:
+            subject_ids = np.arange(len(all_file_paths))
+
+        all_windows_ds, all_infos = [], []
+        for subject_id in subject_ids:
+            file_path = all_file_paths[subject_id]
+            raw = mne.io.read_raw_edf(file_path)
+            path_splits = file_path.split("/")
+            if "abnormal" in path_splits:
+                pathological = True
+            else:
+                assert "normal" in path_splits
+                pathological = False
+            if "train" in path_splits:
+                session = "train"
+            else:
+                assert "eval" in path_splits
+                session = "eval"
+            age, gender = _parse_age_and_gender_from_edf_header(file_path)
+            info = pd.DataFrame(
+                [[age, pathological, gender, session, subject_id]],
+                columns=["age", "pathological", "gender",
+                "session", "subject"], index=[subject_id])
+            base_ds = BaseDataset(raw, info, target=target)
```

Comment:
Need to go through a BaseDataset first?

Reply:
I am not sure if it is actually needed. But, I think we need something addition to the mne Raw object to hold subject_id, session_id, run_id, age, gender, pathology status, or else.

Reply:
Got it! In that case, I think we could find a way for the windower to expect that information, and for its output to also contain that information, either as (1) a structure inside the mne.Epochs, (2) a separate output argument or (3) as part of the metadata DataFrame.

Maybe not essential for now, but it might be cleaner.